### DISCUSSION TITLE
[cloud_firestore] _CodecUtility isn't testable/mockable.

### DISCUSSION THREAD

**Discussion by sensuikan1973** [firebase/flutterfire#6567] [Feature request]
For example, mocking arrayUnion with DocumentReference is impossible.

It's because [_CodecUtility](https://github.com/FirebaseExtended/flutterfire/blob/master/packages/cloud_firestore/cloud_firestore/lib/src/utils/codec_utility.dart) sees [`DocumentReference._deleagte`](https://github.com/FirebaseExtended/flutterfire/blob/8c1a49d0edcbecb688d0f0b69411a954680ed1b9/packages/cloud_firestore/cloud_firestore/lib/src/document_reference.dart#L14).
https://github.com/FirebaseExtended/flutterfire/blob/77b27d042e766b09ba12a680697b6be38f78564d/packages/cloud_firestore/cloud_firestore/lib/src/utils/codec_utility.dart#L46-L47
`delegate` is private. So, we can't mock that.

Ofcourse, delegate should be private. 
So, I wonder if the solution is to fix `_CodecUtility` testable.

**Comment by Ehesp** [comment]
Related; https://github.com/FirebaseExtended/flutterfire/issues/4563
